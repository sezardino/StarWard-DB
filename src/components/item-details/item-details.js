import React from 'react';
import PropTypes from 'prop-types';

import './item-details.css';

const ItemDetails = (props) => {
	const {data, children} = props;
	if (!data) {
		return <p>Choose a character</p>;
	}
	const {name, img} = data;

	return (
		<div className="person-details card">
			<img className="person-image" src={img} alt={name} />

			<div className="card-body">
				<h4>{name}</h4>
				<ul className="list-group list-group-flush">
					{React.Children.map(children, (child) => React.cloneElement(child, {data}))}
				</ul>
			</div>
		</div>
	);
};

const Record = (props) => {
	const {label, field, data} = props;
	return (
		<li className="list-group-item">
			<span className="term">{label}</span>
			<span>{data[field]}</span>
		</li>
	);
};

ItemDetails.propTypes = {
	data: PropTypes.object,
	children: PropTypes.node.isRequired,
};

Record.propTypes = {
	label: PropTypes.string.isRequired,
	field: PropTypes.string.isRequired,
	data: PropTypes.object,
};

export {Record, ItemDetails};
