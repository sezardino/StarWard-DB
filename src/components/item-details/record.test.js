import React from 'react';
import {shallow} from 'enzyme';

import {Record} from './item-details';

describe('Component Record', () => {
	const testData = {
		label: 'Test label',
		field: 'name',
		data: {id: '1', name: 'Test name', someData: 'Test data'},
	};
	const component = shallow(<Record {...testData} />);

	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});
});
