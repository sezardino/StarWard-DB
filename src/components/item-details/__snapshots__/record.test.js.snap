// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Record Component rendered correctly 1`] = `
"<li className=\\"list-group-item\\">
  <span className=\\"term\\">
    Test label
  </span>
  <span>
    Test name
  </span>
</li>"
`;
